tests/std/tests/P2278R4_ranges_const_iterator_machinery/test.compile.pass.cpp

```diff
@@ -0,0 +1,166 @@
+    static_assert(same_as<iter_reference_t<const_iterator<Iter>>, const int&>);
+    static_assert(same_as<const_sentinel<Iter>, basic_const_iterator<Iter>>);
+    static_assert(same_as<iter_reference_t<const_sentinel<Iter>>, const int&>);
+} // namespace test_input_iter
+
+namespace test_prvalue_iter {
+    using TransformView = ranges::transform_view<ranges::ref_view<deque<float>>, int (*)(float)>;
+
+    using Iter = ranges::iterator_t<TransformView>;
+    static_assert(CanIterConstRef<Iter>);
+    static_assert(CanConstIterator<Iter>);
+    static_assert(same_as<iter_const_reference_t<Iter>, int>);
+    static_assert(same_as<const_iterator<Iter>, Iter>);
+    static_assert(same_as<iter_reference_t<const_iterator<Iter>>, int>);
+    static_assert(same_as<const_sentinel<Iter>, Iter>);
+    static_assert(same_as<iter_reference_t<const_sentinel<Iter>>, int>);
+
+    using ConstIter = ranges::iterator_t<const TransformView>;
+    static_assert(CanIterConstRef<ConstIter>);
+    static_assert(CanConstIterator<ConstIter>);
+    static_assert(same_as<iter_const_reference_t<ConstIter>, int>);
+    static_assert(same_as<const_iterator<ConstIter>, ConstIter>);
+    static_assert(same_as<iter_reference_t<const_iterator<ConstIter>>, int>);
+    static_assert(same_as<const_sentinel<ConstIter>, ConstIter>);
+    static_assert(same_as<iter_reference_t<const_sentinel<ConstIter>>, int>);
+} // namespace test_prvalue_iter
+
+namespace test_vector_bool_iter {
+    using Iter = vector<bool>::iterator;
+    static_assert(CanIterConstRef<Iter>);
+    static_assert(CanConstIterator<Iter>);
+    static_assert(same_as<iter_const_reference_t<Iter>, bool>);
+    static_assert(same_as<const_iterator<Iter>, basic_const_iterator<Iter>>);
+    static_assert(same_as<iter_reference_t<const_iterator<Iter>>, bool>);
+    static_assert(same_as<const_sentinel<Iter>, basic_const_iterator<Iter>>);
+    static_assert(same_as<iter_reference_t<const_sentinel<Iter>>, bool>);
+
+    using ConstIter = vector<bool>::const_iterator;
+    static_assert(CanIterConstRef<ConstIter>);
+    static_assert(CanConstIterator<ConstIter>);
+    static_assert(same_as<iter_const_reference_t<ConstIter>, bool>);
+    static_assert(same_as<const_iterator<ConstIter>, ConstIter>);
+    static_assert(same_as<iter_reference_t<const_iterator<ConstIter>>, bool>);
+    static_assert(same_as<const_sentinel<ConstIter>, ConstIter>);
+    static_assert(same_as<iter_reference_t<const_sentinel<ConstIter>>, bool>);
+} // namespace test_vector_bool_iter
+
+// Test standard sentinels
+static_assert(!CanIterConstRef<default_sentinel_t>);
+static_assert(!CanConstIterator<default_sentinel_t>);
+static_assert(same_as<const_sentinel<default_sentinel_t>, default_sentinel_t>);
+static_assert(!CanIterConstRef<unreachable_sentinel_t>);
+static_assert(!CanConstIterator<unreachable_sentinel_t>);
+static_assert(same_as<const_sentinel<unreachable_sentinel_t>, unreachable_sentinel_t>);

```

Comment:
This test should have `int main() {} // COMPILE-ONLY` like other test.compile.pass.cpp, example: https://github.com/microsoft/STL/blob/febb6430ac48a6650b174d567a335afcc0db8dc8/tests/std/tests/P2445R1_forward_like/test.compile.pass.cpp#L72

Reply:
Yes, we (sadly) have different internal and GitHub test runners. The LIT-based runner for GitHub CI knows that `compile.pass.cpp` means "just compile, don't link and execute" but our internal runner does not. (Yes, we want to port the LIT-based runner to use in our internal test runs but we haven't found time yet.)